File: loader/src/main/java/de/timolia/lactea/loader/module/InternalModuleAccess.java
package de.timolia.lactea.loader.module;

import com.google.inject.Injector;
import de.timolia.lactea.loader.inject.ModuleInjector;
import java.util.Objects;

/**
 * @author David (_Esel)
 */
public class InternalModuleAccess {
    public static void setInjector(LacteaModule module, Injector injector) {
        Objects.requireNonNull(injector, "injector");
        ModuleInjector moduleInjector = new ModuleInjector(injector);
        module.setInjector(moduleInjector);
    }

    public static ModuleInjector getInjector(LacteaModule module) {
        return module.getInjector();
    }
}

File: module-api/src/main/kotlin/de/timolia/lactea/module/InternalModuleAccess.kt
package de.timolia.lactea.module

import de.timolia.lactea.inject.LacteaInjector

object InternalModuleAccess {
    fun setInjector(module: LacteaModule, injector: LacteaInjector) {
        module.injector = injector
    }
}